graylog2-server/src/main/java/org/graylog/security/authservice/UserDetails.java

```diff
@@ -43,6 +43,8 @@
 
     public abstract String fullName();
 
+    public abstract boolean isExternal();
```

Comment:
Thanks for the feedback @mpfz0r. Here's some of the thinking behind adding this property here, incase that helps the discussion:

The current definition for external users (as I understand) is that they exist in another system of record and are not permitted to be edited in Graylog. Okta users do not fall into this category, since they can be edited in Graylog. The intention of adding the `isExternal` flag to UserDetails is to allow the `ProvisionerService` to provision users with this flag set to false (the Okta case). The `UserDetails` object is just a means of sending user data into the ProvisionerService.

I did not yet have any plan to add any properties to the `UserImpl` persisted DB object, but happy to look at that further if you think we should identify the Okta users somehow.